src/ocean/Assets.ts

```diff
@@ -239,82 +239,96 @@ export class Assets extends Instantiable {
     return (await this.ocean.metadatacache.queryMetadata(searchQuery)).results
   }
 
+  /**    Metadata updates
+   *  Don't forget to call ocean.OnChainMetadataCache.update after using this functions
+   * ie:  ocean.OnChainMetadataCache.update(ddo.id,ddo,account.getId())
+   */
+
   /**
-   * Edit Metadata for a DDO.
-   * @param  {did} string DID.
+   * Edit Metadata for a DID.
+   * @param  {did} string DID. You can leave this empty if you already have the DDO
+   * @param  {ddo} DDO if empty, will trigger a retrieve
    * @param  {newMetadata}  EditableMetadata Metadata fields & new values.
    * @param  {Account} account Ethereum account of owner to sign and prove the ownership.
-   * @return {Promise<string>}
+   * @return {Promise<DDO>} the new DDO
    */
   public async editMetadata(
     did: string,
-    newMetadata: EditableMetadata,
-    account: Account
+    ddo: DDO,
+    newMetadata: EditableMetadata
   ): Promise<DDO> {
-    const oldDdo = await this.ocean.metadatacache.retrieveDDO(did)
-    let i
-    for (i = 0; i < oldDdo.service.length; i++) {
-      if (oldDdo.service[i].type === 'metadata') {
-        if (newMetadata.title) oldDdo.service[i].attributes.main.name = newMetadata.title
-        if (!oldDdo.service[i].attributes.additionalInformation)
-          oldDdo.service[i].attributes.additionalInformation = Object()
+    if (!ddo) ddo = await this.ocean.metadatacache.retrieveDDO(did)
+    if (!ddo) return null
+    for (let i = 0; i < ddo.service.length; i++) {
+      if (ddo.service[i].type === 'metadata') {
```

Comment:
to prevent excessive nesting of whole blocks like this, it should turn around the logic so the whole block does not need to be wrapped:

```
if (ddo.service[i].type !== 'metadata') return null
```

Reply:
no, it just needs to loop the services and update the metadata.
if somebody is going to publish a ddo with access service first and then metadata, using return will just exit and you end up with un-updated metadata..